app/models/user.rb

```diff
@@ -116,10 +116,10 @@ def dummy_password_for_external_auth
   def change_password(oldpwd, newpwd)
     auth = self.class.authenticator(userid)
     raise MiqException::MiqEVMLoginError, "password change not allowed when authentication mode is #{auth.class.proper_name}" unless auth.uses_stored_password?
-    raise MiqException::MiqEVMLoginError, "old password does not match current password" unless User.authenticate(userid, oldpwd)
-
-    self.password = newpwd
-    self.save!
+    if User.authenticate(userid, oldpwd)
```

Comment:
The current user _is_ authenticated at this point, by chance.

Password changes are handled via this method which is [only called in `UserValidationService#validate_user`](https://github.com/manageiq/manageiq/blob/master/app/services/user_validation_service.rb#L28-L34). The method there should really be broken up :confused: The logic there validates the user [before such a change can occur](https://github.com/manageiq/manageiq/blob/master/app/services/user_validation_service.rb#L18-L21), so this exception is never raised. 

We certainly should keep a re-check in this method (must authenticate to change_password if used in some other context); I notice with the authenticator instance above I should probably just do `if auth.authenticate(userid, oldwd)` to avoid initializing another one uselessly.

Sound ok?